addons/payment/controllers/portal.py

```diff
@@ -175,6 +175,19 @@ def payment_method(self, **kwargs):
         }
         return request.render('payment.payment_methods', rendering_context)
 
+    @http.route('/my/payment_method/archive', type='json', auth='user')
+    def archive_token(self, token_id):
+        user_partner = request.env.user.partner_id
```

Comment:
Guideline tip: append `_sudo` to variable holding sudoed recordset, which is the case for `user_partner` since `request.env.user` is always sudoed.

While we're at it, let's just keep the 'partner' part and then change this to:
```suggestion
        partner_sudo = request.env.user.partner_id
```